### PR TITLE
new: added Meta tags

### LINKED ISSUE #1302: Add meta tags to Expensify.cash html for link previews
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___
View the job on Upwork [here](https://www.upwork.com/jobs/~01a827f3a0a815553e)

**Platform - version:**  
iOS
Android

**Action Performed (reproducible steps):**
1. Copy link to https://expensify.cash
2. Send the link via:
a. iPhone iMessage 
b. iPhone apps: WhatsApp, Facebook, Slack, Twitter
c. Android Text Message
d. Android apps: WhatsApp, Facebook, Slack, Twitter
3. Take note of the link preview that is generated (or not generated)

**Desired Result:**
Add meta tags to Expensify.cash HTML to generate a different selected preview and description (image file and description copy will be provided).

File: 
[expensifycash_metapreviewimage.png.zip](https://github.com/Expensify/Expensify.cash/files/5852544/expensifycash_metapreviewimage.png.zip)

**Actual Result:**
Sharing a link to expensify.cash either doesn't generate a preview (e.g., WhatsApp) or generates a very basic preview (e.g., iMessage)

**Notes/Photos/Videos:** 
Current link preview on iOS/iMessage
image

Current link preview on iOS/WhatsApp
image

Desired link preview (image):
image

Desired link preview (copy):
> Chat and payments are now the same thing. Launching Summer 2021, join the waitlist to be first in line!

### PR DESCRIPTION
**Pull request by parasharrajat**
@marklouisdeshaun Please review.

### Details
Added Meta tags for Link Preview into the `web/index.html` file which renders the desired link preview on social media and other providers.

### Fixed Issues
Fixes  #1302

### Tests
I tested by sending the link on various platforms.

### Instruction to test
After pulling the PR code, You can run a local server of the computer and then use any tunnel provider to host the local server from your computer. I used `ngrok`. In case you are using `ngrok`. You have to change the URL in the 
`og:image` meta tag so that it shows the live URL of the app. For example.
After running ngrok server, you will get a URL like 
```
https://149ac5081107.ngrok.io/
```
now you need to replace the following code 
```
 <meta property="og:image" content="https://expensify.cash/icon-square@4x.png">
<! -- ..... other code --->
 <meta property="twitter:image" content="https://expensify.cash/icon-square@4x.png">
 
 ```
 Into
```
 <meta property="og:image" content="https://149ac5081107.ngrok.io/icon-square@4x.png">
<! -- ..... other code --->
 <meta property="twitter:image" content="https://149ac5081107.ngrok.io/icon-square@4x.png">
 
 ``` 
 After the above steps, Just use the preceding URL and send it on different platforms and observe the new preview.

### Screenshots
Screenshot_2021-01-23
Whatsapp Android.

### CODE REVIEW

**Comment by parasharrajat** [comment]
@marklouisdeshaun I think we need a little heavy image for Whatsapp or maybe my WhatsApp is compressing it.
Can you please confirm is this what you want? Let me know otherwise.

**Comment by shawnborton** [comment]
@parasharrajat happy to provide a larger image for you to try. Here are two options, one is a .jpg and one is a .png:
[Metapreviewimages.zip](https://github.com/Expensify/Expensify.cash/files/5865583/Metapreviewimages.zip)

**Comment by deetergp** [comment]
It sounds like you've made the PR without first proposing a solution. Let's make sure we're doing things in the right order. Also, please be sure to include detailed testing instructions — if I can't test it, I can't merge it.

**Comment by parasharrajat** [comment]
@deetergp Sorry about that. I have proposed a solution to the Main issue and Added few instruction steps in the PR description.

**Comment by parasharrajat** [comment]
@marklouisdeshaun @shawnborton I have tried multiple links on my WhatsApp like youtube.com. But WhatsApp is compressing the image before previewing it. So it looking distorted.
image

What do you say? Should I move to the old image back? which is lesser in size. I suggest we should.

**Comment by shawnborton** [comment]
I agree, the previous image makes sense given that the newly supplied asset does not give us any clear benefit.

**Comment by parasharrajat** [comment]
Rollback to the lighter image(in size). Updated please review.

**Comment by shawnborton** [comment]
A couple of pieces of feedback. 

First, it actually looks like what you are using for the "title" is actually the "description."

Can we do this:
Title: `Expensify.cash`
Description: `Chat and payments are now the same thing. Launching Summer 2021, join the waitlist to be first in line!`

cc @tjferriss, I figured for the title we can keep it simple as `Expensify.cash` unless you have any better ideas. 

Also, it looks like we can remove some of the Twitter-specific tags you are using (notable the `twitter:title` tag) as Twitter works well with the standard Open Graph (og) protocol. [Take a look here (bottom of page)](https://developer.twitter.com/en/docs/twitter-for-websites/cards/guides/getting-started), this would be an ideal set of tags:

```
<meta name="twitter:card" content="summary" />
<meta name="twitter:site" content="@nytimesbits" />
<meta name="twitter:creator" content="@nickbilton" />
<meta property="og:url" content="http://bits.blogs.nytimes.com/2011/12/08/a-twitter-for-my-sister/" />
<meta property="og:title" content="A Twitter for My Sister" />
<meta property="og:description" content="In the early days, Twitter grew so quickly that it was almost impossible to add new features because engineers spent their time trying to keep the rocket ship from stalling." />
<meta property="og:image" content="http://graphics8.nytimes.com/images/2011/12/08/technology/bits-newtwitter/bits-newtwitter-tmagArticle.jpg" />
```

**Comment by parasharrajat** [comment]
@shawnborton. Actually original issue needs this to be the title so I have added that. I will make the changes.

**Comment by shawnborton** [comment]
Let's get rid of `creator` (will update my comment above), and for `site` we can just use `@expensify` - of course, let's test that out first and see how it looks.

**Comment by shawnborton** [comment]
Also - I'm curious what the difference between `summary_large_image` and `summary` is for the `twitter:card` tag.

**Comment by parasharrajat** [comment]
@shawnborton  Check the different values here https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/abouts-cards

**Comment by shawnborton** [comment]
Yeah, they just happen to look identical...

`summary` | `summary_large_image`
----- | ----
image | image

So I say we just go with `summary`

**Comment by parasharrajat** [comment]
@shawnborton  Ok. The main difference is between the image sizes. I am not sure how they can be tested but in `summary_large_image`, We can use it to show the big high-resolution image to get user attention.

I think in our case, I will move to `summary`

**Comment by tjferriss** [comment]
`Expensify.cash` title works for me!

**Comment by parasharrajat** [comment]
Updated. Please review.

File: config/webpack/webpack.common.js
Comment (shawnborton) [inline config/webpack/webpack.common.js:34]: I think we only need one of these, right? Also, could you rename the image file to something more descriptive like `og-preview-image.png`

File: web/index.html
Comment (shawnborton) [inline web/index.html:7]: I think we can get rid of these comments as it's clear what the tags below are for (they say Twitter).

File: web/index.html
Comment (shawnborton) [inline web/index.html:10]: Same, I think we can get rid of this comment.

Comment (parasharrajat) [reply web/index.html:10]: sure

**Comment by parasharrajat** [comment]
Updated.

**Review by shawnborton — APPROVED** [review]

**Comment by shawnborton** [comment]
Looks good to me, thanks! All you @deetergp

**Comment by parasharrajat** [comment]
@shawnborton I think I accidentally re-requested a review from you. 😃

**Review by shawnborton — DISMISSED** [review]

**Review by deetergp — CHANGES_REQUESTED** [review]
The detailed instructions were much better, thanks! I see the image and text in WhatsApp, which is great but I only see the image, without the copy in the Apple Messages app. Since Message is named specifically in the original GH, I think it's important that I see it work before merging. 

### iOS
744EC2C8-F747-43B4-86FC-3EE01098FCA3_1_201_a

### macOS
00ABAEB5-D274-4E3B-B66F-89BB49825890_1_105_c

**Comment by parasharrajat** [comment]
@deetergp  Based on the Changes suggested by @shawnborton  in the comments above, I have added the title as `Expensify.cash`. But Apple messaging app only understands `title` and `Image`, It does not read the `description`. You can see the details [here](https://developer.apple.com/library/archive/technotes/tn2444/_index.html).
@shawnborton  Let me know what should I do here?

**Comment by shawnborton** [comment]
Yeah, I think this is an unfortunate limitation of iMessage really. That's why it might be helpful to have some kind of creative title that has just a little bit of a description. 

Again, looping in our copy guy @tjferriss here, what if we did something like:

Title: `Expensify.cash - Chat with friends and send and receive money`
Description: `Chat and payments are now the same thing. Launching Summer 2021, join the waitlist to be first in line!`

**Comment by deetergp** [comment]
Ah, I see. Well, by comparison, our expensify.com title makes it look a bit more robust in Messages. Let's see what @tjferriss has to see then go with that.
Screen Shot 2021-01-27 at 8 33 16 AM

**Comment by tjferriss** [comment]
Thanks for details. I like the enhanced title. Can we change "and" to "to" in the title so it reinforces they are one in the same, similar to the description.

>Expensify.cash - Chat with friends to send and receive money

**Comment by parasharrajat** [comment]
Ok. Thanks for the feedback guys. I am updating the title now.

**Comment by parasharrajat** [comment]
Updated. Please review.
This is how it looks now.
image

**Comment by shawnborton** [comment]
Looks good to me! I’m away from my laptop for the night so Scott feel free
to merge.

**Review by deetergp — APPROVED** [review]
Looks good, thanks!

**Pull request merged**
